### PR TITLE
client: fixes for extended status

### PR DESCRIPTION
**Pull request by maxtaco**
- Make a `config.HelloIAm()` RPC so that clients can introduce themselves
- Plumb that RPC through to the connection manager
- Have the status command query this table
- Some status fixes:
  - Use Device RPC structure to reuse code
  - Don't crap out on bad provisioning scenarios

### CODE REVIEW

**Comment by maxtaco** [comment]
@patrickxb, can you please take a look? It's not perfect, but maybe let's get it in?

**Comment by patrickxb** [comment]
I'll have to look in more detail in the morning.

**Comment by maxtaco** [comment]
I'll add OS version and Go version too

File: go/libkb/connmgr.go [hunk 1]
```diff
@@ -67,6 +83,41 @@ func (c *ConnectionManager) Shutdown() {
...
+
+type byClientType []keybase1.ClientDetails
+
+func (a byClientType) Len() int           { return len(a) }
+func (a byClientType) Swap(i, j int)      { a[i], a[j] = a[j], a[i] }
+func (a byClientType) Less(i, j int) bool { return a[i].ClientType < a[j].ClientType }
+
+func (c *ConnectionManager) listAllLabeledConnections() (ret []keybase1.ClientDetails) {
+	for _, v := range c.lookup {
+		if d := v.details; d != nil {
```
Comment (patrickxb) [inline go/libkb/connmgr.go]: what's the purpose of d?  can't you just use `v.details`?

File: go/libkb/connmgr.go [hunk 2]
```diff
@@ -21,22 +24,35 @@ type lookupConnectionObj struct {
 	ch chan<- rpc.Transporter
 }
 
+type labelConnectionObj struct {
+	i  ConnectionID
```
Comment (patrickxb) [inline go/libkb/connmgr.go]: `id` instead of `i`?

File: go/service/ctl.go [hunk 3]
```diff
@@ -58,3 +58,7 @@ func (c *CtlHandler) DbNuke(_ context.Context, sessionID int) error {
 	// Now drop caches, since we had the DB's state in-memory too.
 	return c.G().ConfigureCaches()
 }
+
+func (c *CtlHandler) HelloIAm(_ context.Context, arg keybase1.ClientDetails) error {
```
Comment (patrickxb) [inline go/service/ctl.go]: is this leftover code?

**Comment by patrickxb** [comment]
LGTM, minor details mentioned above.

**Pull request merged**